app/components/panel_component/panel.js

```diff
@@ -0,0 +1,40 @@
+import { Controller } from '@hotwired/stimulus';
+import './panel.scss';
+
+export const COMPONENT_CLASS = 'panel-component';
+export const TOGGLE_ELEMENT_CLASS = `${COMPONENT_CLASS}__toggle`;
+export const CONTENT_ELEMENT_CLASS = `${COMPONENT_CLASS}__content`;
+export const CLOSE_ELEMENT_CLASS = `${COMPONENT_CLASS}__content__close-button`;
+export const PANEL_VISIBLE_CLASS = `${CONTENT_ELEMENT_CLASS}--visible`;
+
+export default class extends Controller {
+  static targets = ['toggle', 'content'];
+
+  connect() {
+    this.contentTarget.addEventListener('keydown', (e) => {
+      if (['Esc', 'Escape'].includes(e.key)) {
+        this.contentTarget.classList.remove(PANEL_VISIBLE_CLASS);
+        this.setHiddenState(true);
+      }
+    });
+  }
+
+  toggle() {
+    this.contentTarget.classList.toggle(PANEL_VISIBLE_CLASS) ? this.setVisibleState() : this.setHiddenState(true);
```

Comment:
I'm confused by how `toggle` seems to be calling itself here. Can one of them be renamed?

Reply:
toggle in the second instance is a method of the JS classlist api (toggles a class on an element) so this cant be changed -  i could rename toggle() method i guess

Reply:
It would be good for the method to have a different name from both the JS classlist api and from the target's name.